src/types.ts

```diff
@@ -220,6 +220,61 @@ export type ITick = {
   key?: number;
 };
 
+export class SelectionProxy {
+  private map: Map<string, IPhoto> = new Map<string, IPhoto>();
+
+  add(photo: IPhoto): this {
+    this.map.set(photo.key!, photo);
+    return this;
+  }
+
+  get(photo: IPhoto): IPhoto | undefined {
+    return this.map.get(photo.key!);
```

Comment:
This, precisely, could be the issue. `Photo.key` is nullable because, well, it can be null. The bang operator discards this information without a check. There are indeed a bunch of places where we "fake" a `IPhoto` object that doesn't have a key.

By now though, I'm hoping the selection manager doesn't have to deal with these things. So a runtime assertion here _might_ suffice, so we know if something breaks.